### ISSUE TITLE
Plugin route prefixing doesn't work in 2.0.0-rc.0

### ISSUE THREAD

**Issue by KidkArolis** [fastify/fastify#1279]
## 💥 Regression Report

The { prefix: "/prefix" } option has no effect when used in combination with a `fastify-plugin`.

## Last working version

Worked up to version: latest in series 1.

Stopped working in version: 2.0.0-rc.0

## To Reproduce

Try this:

```
const fastify = require('fastify')();
const fp = require('fastify-plugin');

fastify.get('/foo', function(request, reply) {
  reply.send('hello foo')
});

fastify.register(fp(function (fastify, options, next) {
  fastify.get('/bar', function (request, reply) {
    reply.send('hello prefixed bar')
  })
  next()
}, { prefix: '/prefixed' }))

// listen for requests :)
fastify.listen(process.env.PORT, function() {
  console.log('Your app is listening on port ' + process.env.PORT);
});
```

I have it running over here:
https://glitch.com/edit/#!/roan-echidna

## Expected behavior

Opening `/prefixed/bar` should work, but only `/bar` works.

## Your Environment

I guess it's just a matter of updating `fastify-plugin`? Prefixing works if `fastify-plugin` is not used and a `(fastify) => {}` is registed directly.

**Comment by mcollina** [comment]
prefix never worked with `fastify-plugin`. fastify-plugin skips encapsulation, and prefix will just be a pass-through option.

**Comment by delvedor** [comment]
Are you sure this is working in Fastify v1? It should not.
https://www.fastify.io/docs/latest/Routes/#route-prefixing

**Comment by KidkArolis** [comment]
🤦‍ Ok, sorry for the noise, I must have just gotten confused when migrating some of my app's code into a module. I wrapped it into a plugin, but probably didn't need to.

Been just learning the framework, wrapping my head around the plugins, scoping, hooks and all that. I switched to v2 to use the new error hook (although I realise I could have probably used the old way to do what I needed).

Anyway, closing, and again, sorry for the noise.
